src/opera/pge/disp_s1/disp_s1_pge.py

```diff
@@ -865,3 +884,326 @@ def __init__(self, pge_name, runconfig_path, **kwargs):
+
+        #TODO: Later deliveries will probably have a better way to derive sensor & valid time
+
+        input_file_sample = self.runconfig.sas_config['dynamic_ancillary_file_group']['static_layers_files'][0]
+
+        input_file_fields = basename(input_file_sample).split('_')
+
+        sensor = input_file_fields[5]
+        validity_start_date = input_file_fields[4]
+
+        inter_filename_parts = list(splitext(inter_filename))
+        inter_filename_parts[0] = inter_filename_parts[0].removesuffix('.browse')
+        inter_disp_product_filename = '.'.join(inter_filename_parts[:1] + ["tif"])
+
+        # Check if we've already cached the product metadata corresponding to
+        # this set of intermediate products (there can be multiple sets of
+        # .nc and .png output files when running in historical mode)
+        if inter_disp_product_filename in self._product_metadata_cache:
+            disp_metadata = self._product_metadata_cache[inter_disp_product_filename]
+        else:
+            disp_metadata = self._collect_disp_s1_static_product_metadata(inter_disp_product_filename)
+            self._product_metadata_cache[inter_disp_product_filename] = disp_metadata
+
+        # ProductVersion: OPERA DISP-S1-STATIC product version number with four
+        # characters, including the letter “v” and two digits indicating the
+        # major and minor versions, which are delimited by a period
+        product_version = self.runconfig.sas_config['product_path_group']['product_version']
+
+        if not product_version.startswith('v'):
+            product_version = f'v{product_version}'
+
+        disp_s1_static_product_filename = f"{core_filename}_{frame_id}_{validity_start_date}_{sensor}_{product_version}"
+
+        # Cache the file name for this set of DISP products, so it can be used
+        # with the ISO metadata later.
+        self._product_filename_cache[inter_disp_product_filename] = disp_s1_static_product_filename
+
+        return disp_s1_static_product_filename
+
+    def _geotiff_filename(self, inter_filename):
+        """
+        Returns the file name to use for GeoTIFF's produced by the DIST-S1-STATIC PGE.
+
+        The GeoTIFF filename for the Base PGE consists of:
+
+            <Core filename>.tif
+
+        Where <Core filename> is returned by DispS1StaticPostProcessorMixin._core_filename()
+
+        Parameters
+        ----------
+        inter_filename : str
+            The intermediate filename of the output GeoTIFF to generate
+            a filename for. This parameter may be used to inspect the file
+            in order to derive any necessary components of the returned filename.
+
+        Returns
+        -------
+        geotiff_filename : str
+            The file name to assign to GeoTIFF product(s) created by this PGE.
+        """
+
+        # I think there should be a better way to do this. The SAS outputs just the band names
```

Comment:
Let's through a TODO in front of this to help track it. We can probably ask Scott S. to change the default output names he uses to better accommodate this.

Reply:
> We can probably ask Scott S. to change the default output names he uses to better accommodate this.

That's what I was thinking. I think the best solution would be to change the layer names to use `-` instead of `_`